### PR TITLE
Makefile: add fmt, update help

### PR DESCRIPTION
**Pull request by tr1sm0s1n**
- Added `fmt` command to **Makefile** for easier formatting.
- Formatted '.go' files with `make fmt`.
- Made `help` the default command and updated it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,5 +51,11 @@ devtools:
...
 help: Makefile
-	@echo " Choose a command run in go-ethereum:"
+	@echo ''
+	@echo 'Usage:'
+	@echo '  make [target]'
+	@echo ''
+	@echo 'Targets:'
 	@sed -n 's/^#?//p' $< | column -t -s ':' |  sort | sed -e 's/^/ /'
+
+.DEFAULT_GOAL := help
```
Comment (fjl) [inline Makefile:61]: help should not be the default. the default is to build.

File: Makefile [hunk 2]
```diff
@@ -2,39 +2,45 @@
...
 # don't need to bother with make.
 
-.PHONY: geth all test lint clean devtools help
+.PHONY: geth all test lint fmt clean devtools help
 
 GOBIN = ./build/bin
 GO ?= latest
 GORUN = go run
+GOFMT ?= gofmt "-s"
+GOFILES := $(shell find . -name "*.go")
```
Comment (fjl) [inline Makefile:11]: It's not necessary to define these as variables.

**Review by fjl — CHANGES_REQUESTED** [review]
Please undo changes in Go files.

File: Makefile (hunk 1 above)
Comment (tr1sm0s1n) [reply Makefile]: Fixed in [c8e205a](https://github.com/ethereum/go-ethereum/pull/29777/commits/c8e205a5e46f51e1e66bd09e9308fc1fb3cbb731).

File: Makefile (hunk 2 above)
Comment (tr1sm0s1n) [reply Makefile]: Fixed in [c8e205a](https://github.com/ethereum/go-ethereum/pull/29777/commits/c8e205a5e46f51e1e66bd09e9308fc1fb3cbb731).

**Pull request merged**